examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,193 @@
+# %%
+# Compared to the OLS estimator, the coefficients using a Bayesian Ridge
+# regression are slightly shifted toward zeros, which stabilises them. The ARD
+# regression provides a sparser solution: some of the non-informative
+# coefficients are set exactly to zero, while shifting others closer to zero.
+# Some non-informative coefficients are still present and retain the order of
+# magnitude of the ground truth coefficients. Due to the added noise, none of
+# the models recover the true weights. Indeed, all models always have more than
+# 10 non-zero coefficients.
 
 # %%
-# Generating simulated data with Gaussian weights
-
-# Parameters of the example
-np.random.seed(0)
-n_samples, n_features = 100, 100
-# Create Gaussian data
-X = np.random.randn(n_samples, n_features)
-# Create weights with a precision lambda_ of 4.
-lambda_ = 4.0
-w = np.zeros(n_features)
-# Only keep 10 weights of interest
-relevant_features = np.random.randint(0, n_features, 10)
-for i in relevant_features:
-    w[i] = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(lambda_))
-# Create noise with a precision alpha of 50.
-alpha_ = 50.0
-noise = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(alpha_), size=n_samples)
-# Create the target
-y = np.dot(X, w) + noise
+# Plot the marginal log-likelihood
+# --------------------------------
+import numpy as np
+
+ard_scores = -np.array(ard.scores_)
+brr_scores = -np.array(brr.scores_)
+plt.plot(ard_scores, color="navy", label="ARD")
+plt.plot(brr_scores, color="red", label="BayesianRidge")
+plt.ylabel("Log-likelihood")
+plt.xlabel("Iterations")
+plt.xlim(1, 30)
+plt.legend()
+_ = plt.title("Models log-likelihood")
 
 # %%
-# Fit the ARD Regression
-clf = ARDRegression(compute_score=True)
-clf.fit(X, y)
+# In this case ARD shows a faster convergence than BayesianRidge.
+#
+# Bayesian regressions with polynomial feature expansion
+# ======================================================
+# Generate synthetic dataset
+# --------------------------
+# We create a target that is a non-linear function of the input feature.
+# Noise following a standard uniform distribution is added.
+
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import PolynomialFeatures, StandardScaler
+
+rng = np.random.RandomState(0)
+n_samples = 110
+
+# sort the data to make plotting easier later
+X = np.sort(-10 * rng.rand(n_samples) + 10)
+noise = rng.normal(0, 1, n_samples) * 1.35
+y = np.sqrt(X) * np.sin(X) + noise
+full_data = pd.DataFrame({"input_feature": X, "target": y})
+X = X.reshape((-1, 1))
 
-ols = LinearRegression()
-ols.fit(X, y)
+# %%
+# Fit the regressors
+# ------------------
+#
+# Here we try a degree 10 polynomial to potentially overfit, though the bayesian
+# linear models regularize the size of the polynomial coefficients.
+# As `fit_intercept=True` by default for `ARDRegression()` and `BayesianRidge()`,
+# then `PolynomialFeatures` should not introduce an additional bias feature.
+
```

Comment:
I think that we can add an additional comment regarding that the predictions that we get with the models are not only the mean but the standard deviation because we are using a Bayesian approach.